osclib/stagingapi.py

```diff
@@ -1581,27 +1616,49 @@ def create_adi_project(self, name):
         if name in adi_projects:
             raise Exception('Project {} already exist'.format(name))
 
-        meta = """
-        <project name="{0}">
-          <title></title>
-          <description></description>
-          <url>/project/staging_projects/{1}/adi:{2}</url>
-          <publish>
-            <disable/>
-          </publish>
-          <debuginfo>
-            <enable/>
-          </debuginfo>
-          <repository name="standard">
-            <path project="{1}" repository="standard"/>
-            <arch>x86_64</arch>
-          </repository>
-        </project>""".format(name, self.project, self.extract_adi_number(name))
+        if use_frozenlinks:
+            meta = """
+            <project name="{0}">
+              <title></title>
+              <description></description>
+              <link project="{1}"/>
```

Comment:
If this is the only line different surely complete duplication of the string can be avoided. Either parse as XML and add element, or add a string placeholder.
```
'<description></description>{}'.format("<link project="{1}"/>\n")
```
etc